tests/sql/test_exprs.py

```diff
@@ -135,3 +138,82 @@ def test_is_in_edge_cases():
     # Test with mixed types in the IN list
     with pytest.raises(Exception, match="All literals must have the same data type"):
         daft.sql("SELECT * FROM df WHERE nums IN (1, '2', 3.0)").collect().to_pydict()
+
+
+@pytest.mark.parametrize(
+    "date_values, ts_values, expected_intervals",
+    [
+        # Adjust expected intervals to match actual datetime and date object formats
+        (
+            ["2022-01-01", "2020-02-29", "2019-05-15"],
+            ["2022-01-01 10:00:00", "2020-02-29 23:59:59", "2019-05-15 12:34:56"],
+            {
+                "date_add_day": [
+                    datetime.date(2022, 1, 2),
+                    datetime.date(2020, 3, 1),
+                    datetime.date(2019, 5, 16),
+                ],
+                "date_sub_year": [
+                    datetime.date(2021, 1, 1),
+                    datetime.date(2019, 3, 1),  # datetime.date(2019, 2, 28)
+                    datetime.date(2018, 5, 15),
+                ],
+                "ts_add_hour": [
+                    datetime.datetime(2022, 1, 1, 11, 0, 0),
+                    datetime.datetime(2020, 3, 1, 0, 59, 59),
+                    datetime.datetime(2019, 5, 15, 13, 34, 56),
+                ],
+                "ts_sub_minute": [
+                    datetime.datetime(2022, 1, 1, 9, 59, 0),
+                    datetime.datetime(2020, 2, 29, 23, 58, 59),
+                    datetime.datetime(2019, 5, 15, 12, 33, 56),
+                ],
+            },
+        ),
+    ],
+)
+def test_interval_comparison(date_values, ts_values, expected_intervals):
+    # Create DataFrame with date and timestamp columns
+    df = daft.from_pydict({"date": date_values, "ts": ts_values}).select(
+        col("date").cast(daft.DataType.date()), col("ts").str.to_datetime("%Y-%m-%d %H:%M:%S")
+    )
+    catalog = SQLCatalog({"test": df})
+
+    # Define interval function for DataFrame operations
+    def interval(unit, multiplier=1):
+        if unit == "year":
+            td = datetime.timedelta(days=365 * multiplier)
+        else:
+            td = datetime.timedelta(**{unit: multiplier})
+        total_microseconds = int(td.total_seconds() * 1_000_000)
+        return daft.lit(total_microseconds).cast(daft.DataType.duration("us"))
+
+    expected_df = (
+        df.select(
+            (col("date") + interval("days", 1)).alias("date_add_day"),
+            (col("date") - interval("year", 1)).alias("date_sub_year"),
+            (col("ts") + interval("hours", 1)).alias("ts_add_hour"),
+            (col("ts") - interval("minutes", 1)).alias("ts_sub_minute"),
+        )
+        .collect()
+        .to_pydict()
+    )
+
+    actual_sql = (
+        daft.sql(
+            """
+        SELECT
```

Comment:
Does this handle composite intervals? `12 year 11 months 30 days`. It's fine if it doesn't at this point, but we should atleast make sure it doesn't panic.

Reply:
Sure thing, added a regex to parse `IntervalPart`